lib/Analysis/Utility.cpp

```diff
@@ -486,6 +486,14 @@ struct DFSState {
         ready = false;
       }
     }
+    Operation *parent = op->getParentOp();
```

Comment:
Why does it have to visit parent ops? Is there an example?

Reply:
This is to get the right order when using SCF ops. For instance if you have 3 ops like this:
```
%0 = triton.op1
%1 = scf.for(...)
   ...
   %2 = triton.op2
   scf.yield
```
you would want the order to be %0, %1, %2. Without this the topological order could be %0, %2, %1

Reply:
OK, I thought it could be handled by postmortem analysis.

Could you please leave a comment in the code to denote the purpose?

Reply:
done.